src/quicklist.c

```diff
@@ -794,9 +794,10 @@ void quicklistReplaceEntry(quicklistIter *iter, quicklistEntry *entry,
             unsigned char *p = lpSeek(entry->node->entry, -1);
             quicklistDelIndex(quicklist, entry->node, &p);
             entry->node->dont_compress = 0; /* Re-enable compression */
-            _quicklistMergeNodes(quicklist, entry->node);
-            quicklistCompress(quicklist, entry->node);
-            quicklistCompress(quicklist, entry->node->next);
+            new_node = _quicklistMergeNodes(quicklist, new_node);
+            quicklistCompress(quicklist, new_node);
+            quicklistCompress(quicklist, new_node->prev);
+            if (new_node->next) quicklistCompress(quicklist, new_node->next);
```

Comment:
i'm a little bit confused about what quicklistCompress does.
maybe you can take this opportunity to extend the documentation about it, and the functions it calls.

Reply:
i'm also confused about whethere or not to use `quicklistCompress()`, now that we need to use the context to  determine to compress or not.
In face, `quicklistCompress` doesn't have good performance, we need to traverse the entire quicklist nodes if
one node is in the mid of quicklist.
i'm thinking about another optimization, we can mark all the nodes need to be comrpess with a flag, and then
uncompress the nodes outside of `quicklist->compress` and reset tthe flag, and then we can directly compress
the nodes whose flag has not been reset.
i'll mark a separate PR to do this.